### DISCUSSION TITLE
[DONE] Nice to have: Add company logo

### DISCUSSION THREAD

**Discussion by infra-monkey** [baimard/gestion#192] [Ideas]
Hello,

I am currently thinking about using this application.
For the moment I find all the required stuff I need to invoice to my clients. Therefore it is a viable simple solution.

I was very happy to see the announcement of such an app.

In order to make the invoice a bit more pretty and "professional" it would be nice to be able to add the company logo to the quotes and invoices.

Obviously, it requires to add that field in the company configuration panel and use it when generating the quotes and invoices.
And of course, the logo should be optional because not every small company has a logo.
I think this would increase the overall user experience.

Unfortunately, I don't have the skillset to contribute directly in the code. However, this doesn't look like a huge work (speaking as a non developper)

What do you think?

**Comment by baimard** [comment]
It's in roadmap ! I need time to code this features. Not very complicated but I need time to implement !

This idea is a real need for all freelancer.

In the future I will also made possible to modify the template of invoice and quote !

**Comment by infra-monkey** [comment]
There's no rush. As said in the title, it's a nice to have feature.
I probably need to go through the existing issues and discussions to get a better view of the project. 

But at the moment I'm quite pleased with what I see. 

Modifying templates will be nice indeed. 

I'm familiar with open source projects so no pressure from me. I understand that you (and other contributors) can't spend a lot of time on it.

**Reply by baimard** [reply]
Yes no pressure, what's incredible is the popularity of this project. That was not expected! 

But I try to do my best, and I'm powered by all the good ideas and the good comment on the application.

**Reply by infra-monkey** [reply]
That means that there is a real need :smile:
And that the implementation is good !

**Comment by baimard** [comment]
Like that ? :)

pdf :
![Capture d’écran de 2022-02-24 19-37-12](https://user-images.githubusercontent.com/7591697/155586521-e4e06741-5fa7-4024-91fe-5d4b4c385f7c.png)

Html :
![Capture d’écran de 2022-02-24 19-35-58](https://user-images.githubusercontent.com/7591697/155586525-149f5853-7802-4686-9a7e-7cf43a8fda15.png)

If you dont have set your logo :
![Capture d’écran de 2022-02-24 19-34-25](https://user-images.githubusercontent.com/7591697/155586526-4da33149-bc39-4988-8eff-3344798b0cbb.png)

**Reply by infra-monkey** [reply]
That was fast, wow !
I think it should just be empty if no logo is defined.
Will the logo be auto rescaled of should the file be scaled before hand? Just an implementation detail.

But it looks good and more personalized :smile:

**Reply by baimard** [reply]
> Will the logo be auto rescaled of should the file be scaled before hand? Just an implementation detail.

Auto rescaled but you still need to be careful of the weight of the file.

this is the last part of my work, where to set the brand logo and recommandations

**Reply by baimard** [reply]
And also that was very fast because I think it's a real need for users !

**Reply by infra-monkey** [reply]
Thank you

**Reply by baimard** [reply]
Soo what do you think about this feature ?

**Reply by infra-monkey** [reply]
I like it, it gives a more professional look
